specs/vmgenid.md

```diff
@@ -0,0 +1,126 @@
+
+| Offset | Field | Description |
+|--------|-------|-------------|
+| 0x00 | `uint64_t generation_id_low` | Lower 64 bits of the 128-bit generation ID |
+| 0x08 | `uint64_t generation_id_high` | Upper 64 bits of the 128-bit generation ID |
+
+The generation ID is a 128-bit cryptographically random value that is unique across all VM instances and time. All 128 bits are random; it is *not* a Version 4 UUID.
+
+The generation ID changes whenever the VM is restored to an earlier or non-unique state:
+
+  - Snapshot restoration
+  - Backup recovery
+  - VM cloning/copying/import
+  - Disaster recovery failover
+
+The generation ID remains constant during normal VM operations:
+
+  - Pause/resume
+  - Shutdown/restart/reboot
+  - Host reboot or upgrade
+  - Live migration or lossless online failover
+
+Events such as live migration which merely disrupt the VM's clock without changing the uniqueness of its identity do not result in a change to the generation ID. Conversely, cloning (forking) a running VM running on the same host would result in a new generation ID without disrupting the timekeeping. Guests which want to detect clock disruption should use the [VMClock device](vmclock.md) for that purpose.
+
+### GUID interoperability
+
+If the generation ID is represented as a GUID for the purpose of storage or configuration by a Virtual Machine Monitor, it is recommended that:
+
+- The generation ID shared to the guest is the little-endian representation of that GUID
+- The textual representation of the GUID, in display or configuration, is the RFC 4122 standard big-endian form
+
+
+## Discovery via ACPI
+
+To expose VMGenID to the operating system via ACPI, the firmware or hypervisor must:
+
+1. Place the shared `vmgenid_abi` structure somewhere in RAM, ROM or device memory space, which is guaranteed not to be used by the operating system. It must not be in ranges reported as `AddressRangeMemory` or `AddressRangeACPI`, and must not be in the same page as any memory which is expected to be mapped by a page table entry with caching disabled.
+
+2. Expose a device somewhere in the ACPI namespace with:
+   - a hardware ID (`_HID`) that is hypervisor-specific
+   - a DOS Device Name ID (`_DDN`) of "VM_Gen_Counter"
+   - a compatible ID (`_CID`) of "VM_Gen_Counter"
+
+3. Attach to the device an "ADDR" method which when evaluated returns the 64-bit physical address of the generation ID structure as a package containing the low and high 32-bit address components in that order.
+
+4. After the generation ID changes, the device shall raise an ACPI Notify operation using notification code 0x80. The device may raise the notify operation even if the generation ID has not changed.
+
+## Discovery via Device Tree
+
+The BIOS or hypervisor must place the `vmgenid_abi` structure in an otherwise unused region of physical memory and advertise its presence to the operating system. The Device Tree binding for the `microsoft,vmgenid` node is as follows:
```

Comment:
```suggestion
The firmware or hypervisor must place the `vmgenid_abi` structure in an otherwise unused region of physical memory and advertise its presence to the operating system. The Device Tree binding for the `microsoft,vmgenid` node is as follows:
```
the same is also still in `vmclock.md`